tests/GenericListCollectionTest.php

```diff
@@ -178,6 +178,16 @@ public function testMap(): void
 		$this->assertSame($value, $newCollection->first()?->value);
 	}
 
+	public function testMapOnEmptyCollection(): void
+	{
+		$collection1 = new GenericListCollection(RandomEntity::class, []);
+
+		$newCollection = $collection1->map(fn (RandomEntity $entity): ExtraEntity => new ExtraEntity((string)$entity->value));
+
+		$this->assertNotSame($collection1, $newCollection);
+		$this->assertCount(0, $newCollection);
+	}
```

Comment:
Would love to see a check to make sure the type has not changed (also for the map test):
```suggestion
		$this->assertSame($collection1->getType(), $newCollection->getType());
	}
```

Reply:
Added 

Also added a comment explaining why it stays the same